kernel/arch/dreamcast/kernel/init.c

```diff
@@ -225,16 +223,18 @@ int arch_main() {
     /* Do auto-init stuff */
     arch_auto_init();
 
-    /* Run ctors */
     __verify_newlib_patch();
+
+    /* Run ctors */
     _init();
 
     /* Call the user's main function */
     rv = main(0, NULL);
 
     /* Call kernel exit */
-    arch_exit();
+    exit(rv);
 
+    /* should never actually return here */
     return rv;
```

Comment:
Probably should get rid of the return with a value, since the function returns void now. Even though it shouldn't ever get there...
```suggestion
    /* should never actually return here */
```

Reply:
Good catch, btw,